### PR TITLE
Feature/linkta 325 set up testing environment for unit testing with jest

### PR DESCRIPTION
**Pull request by kinskiwu**
## Description
This pull request implements the foundational setup for unit testing within our project using Jest and React Testing Library. Key changes include:

- Added necessary packages to enable Jest testing, focusing on unit tests.
- Created and configured Jest-related files to tailor the testing environment.
- Updated `tsconfig.json` to include types specific to testing frameworks and to exclude `cypress.config.ts`, resolving type errors with React Testing Library.
- Introduced `jest-unit.yml` to handle Jest unit tests within our CI/CD pipeline.
- Modified the `package.json` to include new scripts for running Jest tests, ensuring they are integrated into our development workflow.
- Added a simple unit test using React Testing Library to demonstrate the setup's functionality. Note: A unit test for a validator helper was also conducted but is not included in this PR as the helper is being phased out in a subsequent PR.

Fixes #325

## Type of change

- [X] ✨ New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [X] unit test
- [X] e2e test

* Ran `npm run test` & `npm run dev` followed by `npm run cypress-run` to confirm that all existing tests pass without any issues, ensuring that the setup is correctly integrated and functional.

## Checklist:

- [X] I have performed a self-review of my own code.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [X] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [X] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [X] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: tsconfig.json
Comment (kinskiwu) [inline tsconfig.json:1]: The combination of adding all the types in a types array and adding cypress.config.ts in the exclude array was the only combo that I've tested that doesn't cause type errors in either react-testing-library or cypress tests. 

I've also tried the following but still getting the errors:
- rolling back @testing-library/jest-dom and next.js to earlier versions (v5x & v13x) to test package compatibility
- breaking the tsconfig into 3 parts, one for main, one for jest, and one for cypress

File: .github/jest-unit.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  run-jest-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
```
Comment (kinskiwu) [inline .github/jest-unit.yml]: Match the Node.js version with other workflows

File: .github/jest-unit.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Jest Unit Tests
+
+on:
+  pull_request:
```
Comment (kinskiwu) [inline .github/jest-unit.yml]: Only trigger on pull requests to the dev branch - consistent with how the cypress yml is set up

**Review by martin-carmona — APPROVED** [review]

**Review by colinvandergraaf — APPROVED** [review]
LGTM!  I was able to run both jest tests and cypress e2e tests locally.

**Review by ejwoods — COMMENTED** [review]
neat 👍

**Pull request merged**